### PR TITLE
LNX-124 Add pushing objects

### PR DESCRIPTION
**Pull request by filip-szweda**
* `Push` receives a push direction
* it checks if there are any pushable `Object`s in this direction (`pushable` flag, `False` by default)
* if so, it applies `Move` to them and saves their ids
* new unit tests

### CODE REVIEW

File: lynx/common/actions/push.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def apply(self, scene: 'Scene') -> None:
+        target_position = scene.get_object_by_id(self.object_id).position + self.direction
+        objects_on_square = scene.get_objects_by_position(target_position)
+        for pushable_object in filter(lambda x: x.pushable, objects_on_square):
+            Move(object_id=pushable_object.id, movement=self.direction).apply(scene)
+            self.pushed_object_ids.append(pushable_object.id)
+
+    def satisfies_requirements(self, scene: 'Scene') -> bool:
+        target_position = scene.get_object_by_id(self.object_id).position + self.direction
+        return CommonRequirements.is_in_range(scene, self.object_id, target_position, 1) and CommonRequirements.is_pushable(scene, target_position)
```
Comment (Krystian030) [inline lynx/common/actions/push.py]: Do you consider the case if the square where we want to push our object is not available or there is some `no pushable` object?

File: tests/push_test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        assert serialized_push == self.expected_serialized_push
+
+    def test_failure(self) -> NoReturn:
+        push = Push(object_id=456, direction=Direction.EAST.value)
+        push.pushed_object_ids = [4, 5, 6]
+        serialized_push = push.serialize()
+        assert serialized_push != self.expected_serialized_push
+
+
+class TestMoveDeserialization:
```
Comment (Krystian030) [inline tests/push_test.py]: TestMoveDeserialization -> TestPushDeserialization

**Review by Krystian030 — CHANGES_REQUESTED** [review]
Look nice!

File: lynx/common/actions/push.py (hunk 1 above)
Comment (filip-szweda) [reply lynx/common/actions/push.py]: Good point!
I did not think about such a case.

**Pull request merged**